### PR TITLE
test_harness w/ fixes for the unit tests to pass

### LINKED ISSUE #49: add unit tests and test harness
kustomize has a test harness that we should leverage and automate test case generation

### PR DESCRIPTION
**Pull request by kkasravi**
fixes #49

Provides a basic unit test generator that generates golang code for each kustomize target. The golang test cases are then compiled and run by `go test -v .`
- [x] tests are generated and run by running `make` under tests
- [x] `make` first runs hack/gen-test-targets.sh which creates golang code that uses a local copy of [kusttarget_test.go](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/target/kusttarget_test.go)
- [x] after code generation `go test -v .` is called
- [x] certain tests can be skipped by defining EXCLUDE in tests/Makefile
- [x] does not test different values of parameters that are used by most targets (follow-up PR)

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/50)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,70 +2,111 @@
...
+            ├── 🎯cluster-gangscheduled
+            ├── 🎯namespaced
+            └── 🎯namespaced-gangscheduled
 ```
 
 ## Kfctl Processing 
-Kfctl will traverse these directories to find and build kustomize targets based on the configuration file `app.yaml`. App.yaml is derived from a file in the kubeflow [config](https://github.com/kubeflow/kubeflow/tree/master/bootstrap/config) directory. Each target processed by kfctl will result in an output yaml file. The output file is generated by calling kustomize's API.  The kustomize package manager in kfctl will read app.yaml and apply the packages, components and componentParams to kustomize in the following way:
+Kfctl traverses directories under manifests to find and build kustomize targets based on the configuration file `app.yaml`. App.yaml is derived from a file in the kubeflow [config](https://github.com/kubeflow/kubeflow/tree/master/bootstrap/config) directory. Each target processed by kfctl will result in an output yaml file. The output file is generated via kustomize's API. The kustomize package manager in kfctl will read app.yaml and apply the packages, components and componentParams to kustomize in the following way:
```
Comment (hougangliu) [inline README.md]: s/App.yaml/app.yaml

File: README.md [hunk 2]
```diff
@@ -74,10 +115,74 @@ namespace:
 - **component parameters** 
   - are applied to a component's params.env file. There must be an entry whose key matches the component parameter. The params.env file is used to generate a ConfigMap. Entries in params.env are resolved as kustomize vars or referenced in a deployment or statefulset env section in which case no var definition is needed.
 
+Multiple overlays -
 
-Generating yaml output for any target can be done in the following way:
+Kfctl has the capability to combine more than one overlay during `kfctl generate ...`. An example is shown below where the profiles target in [manifests](https://github.com/kkasravi/manifests/tree/multiple_overlays/profiles) can include either debug changes in the Deployment or Device information in the Namespace (the devices overlay is not fully integrated with the Profile-controller at this point in time and is intended as an example) or **both**.
```
Comment (hougangliu) [inline README.md]: please link manifests to https://github.com/kubeflow/manifests/tree/master/profiles

**Comment by hougangliu** [comment]
/lgtm
thanks!

**Comment by jlewi** [comment]
@hougangliu @kkasravi one of you needs to approve this for it to get merged.

**Pull request merged**